src/App.tsx

```diff
@@ -14,11 +15,132 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[],
+  isListOpened: boolean,
+  isReversed: boolean,
+  isSortedByLength: boolean,
+  isSortedByAlphabet: boolean,
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: goodsFromServer,
+    isListOpened: false,
+    isReversed: false,
+    isSortedByLength: false,
+    isSortedByAlphabet: false,
+  };
+
+  listOpener = () => {
+    this.setState(() => ({
+      isListOpened: true,
+    }));
```

Comment:
если не юзаешь предыдущий стейт, функциональный параметр не нужен
```suggestion
    this.setState({
      isListOpened: true,
    });
```